sw/device/silicon_creator/lib/cert/asn1_unittest.cc

```diff
@@ -229,6 +229,58 @@ TEST(Asn1, PushStrings) {
   EXPECT_EQ_CONST_ARRAY(buf, out_size, kExpectedResult);
 }
 
+// Make sure that we can push bitstrings.
+TEST(Asn1, PushBitString) {
+  asn1_state_t state;
+  uint8_t buf[20];
+  EXPECT_EQ(asn1_start(&state, buf, sizeof(buf)), kErrorOk);
+  asn1_bitstring_t bitstring;
+  asn1_tag_t tag;
+  // Empty bitstring.
+  EXPECT_EQ(asn1_start_tag(&state, &tag, kAsn1TagNumberBitString), kErrorOk);
+  EXPECT_EQ(asn1_start_bitstring(&state, &bitstring), kErrorOk);
+  EXPECT_EQ(asn1_finish_bitstring(&bitstring), kErrorOk);
+  EXPECT_EQ(asn1_finish_tag(&tag), kErrorOk);
+  // Two trivial bitstrings.
+  EXPECT_EQ(asn1_start_tag(&state, &tag, kAsn1TagNumberBitString), kErrorOk);
+  EXPECT_EQ(asn1_start_bitstring(&state, &bitstring), kErrorOk);
+  EXPECT_EQ(asn1_bitstring_push_bit(&bitstring, true), kErrorOk);
+  EXPECT_EQ(asn1_finish_bitstring(&bitstring), kErrorOk);
+  EXPECT_EQ(asn1_finish_tag(&tag), kErrorOk);
+
+  EXPECT_EQ(asn1_start_tag(&state, &tag, kAsn1TagNumberBitString), kErrorOk);
+  EXPECT_EQ(asn1_start_bitstring(&state, &bitstring), kErrorOk);
+  EXPECT_EQ(asn1_bitstring_push_bit(&bitstring, false), kErrorOk);
+  EXPECT_EQ(asn1_finish_bitstring(&bitstring), kErrorOk);
+  EXPECT_EQ(asn1_finish_tag(&tag), kErrorOk);
+  // A longer bitstring.
+  const bool kBitString[] = {
```

Comment:
It's a bit inconvenient to have to specify the size in the array type, is there a way to have this automatically deduced? Otherwise I am not sure it's really adding anything versus a plain C array.